pennylane_lightning/core/src/simulators/lightning_kokkos/gates/BasicGateFunctors.hpp

```diff
@@ -621,7 +929,7 @@ void applyIsingZZ(Kokkos::View<Kokkos::complex<PrecisionT> *> arr_,
     const Kokkos::complex<PrecisionT> shift_0 = Kokkos::complex<PrecisionT>{
         std::cos(angle / 2),
         (inverse) ? std::sin(angle / 2) : -std::sin(angle / 2)};
-    const Kokkos::complex<PrecisionT> shift_1 = conj(shift_0);
+    const Kokkos::complex<PrecisionT> shift_1 = Kokkos::conj(shift_0);
```

Comment:
@AmintorDusko When I was reviewing I noticed that some functions have this namespace then I checked the [documentation](https://kokkos.org/kokkos-core-wiki/API/core/numerics/mathematical-functions.html#notes) this suggestion to use explicit qualification `Kokkos::[math_func]` but now we have the question if this only valid for all the code or just for Kokkos parallel region. :thinking: 

I suggest using it in all math calls because Kokkos will be aware of which math library link, e.g. [AMD Math Lib](https://www.amd.com/en/developer/aocl/libm.html)

Reply:
It's mostly useful for parallel region. But I wonder if it adds any abstraction overhead to the code base if we replace all these math ops with Kokkos::func :thinking:

Reply:
Feel free to resolve this conversation when sensible.

Reply:
So I think for parallel region I will use `Kokkos::` for all func, and for non-parallel region I will use `std::` for standard (float/double) operations, and `Kokkos::` for operations on `kokkos::complex` types